File: Java/121.java
class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        for (int i = 0; i < prices.length - 1; i++) {
            if (prices[i] > prices[i + 1]) {
                continue;
            }
            int win = i + 1;
            while (win < prices.length && prices[i] <= prices[win]) {
                result = Math.max(result, prices[win] - prices[i]);
                win++;
            }
            i = win - 1;
        }
        return result;
    }
}

File: Python/121.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        result = 0
        cur_min = prices[0]

        for i in range(len(prices)):
            if prices[i] < cur_min:
                cur_min = prices[i]
            result = max(result, prices[i] - cur_min)

        return result
